### PR TITLE
[IMP] fieldservice_account_analytic: Filter Location/Contact

### PR DESCRIPTION
**Pull request by osi-scampbell**
Add a boolean on FSM Settings to reverse the contact/location filtering that we do on FSM Orders

### CODE REVIEW

File: fieldservice_account_analytic/models/fsm_order.py [hunk 1]
```diff
@@ -25,13 +25,22 @@ def _compute_total_cost(self):
...
-    @api.onchange('location_id')
+    @api.onchange('location_id', 'customer_id')
     def _onchange_location_id_customer_account(self):
-        if self.location_id:
-            return {'domain': {'customer_id': [('service_location_id', '=',
-                                                self.location_id.name)]}}
+        if self.env.user.company_id.filter_location_by_contact:
+            if self.location_id:
+                return {'domain': {'customer_id': [('service_location_id', '=',
+                                                    self.location_id.name)]}}
```
Comment (dreispt) [inline fieldservice_account_analytic/models/fsm_order.py]: Should this be `self.location_id.id`?

File: fieldservice_account_analytic/models/fsm_order.py [hunk 2]
```diff
@@ -25,13 +25,22 @@ def _compute_total_cost(self):
...
+            else:
+                return {'domain': {'customer_id': [('id', '!=', None)],
+                                   'location_id': [('id', '!=', None)]}}
```
Comment (dreispt) [inline fieldservice_account_analytic/models/fsm_order.py]: Doesn't just using empty domains `[]` work?

File: fieldservice_account_analytic/models/fsm_order.py (hunk 1 above)
Comment (dreispt) [reply fieldservice_account_analytic/models/fsm_order.py]: Ping @osi-scampbell

File: fieldservice_account_analytic/models/fsm_order.py (hunk 2 above)
Comment (dreispt) [reply fieldservice_account_analytic/models/fsm_order.py]: Ping @osi-scampbell

File: fieldservice_account_analytic/models/res_company.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Copyright (C) 2019 - TODAY, Open Source Integrators
```
Comment (max3903) [inline fieldservice_account_analytic/models/res_company.py]:
```suggestion
# Copyright (C) 2020 Open Source Integrators
```

File: fieldservice_account_analytic/models/res_company.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ResCompany(models.Model):
+    _inherit = 'res.company'
+
+    filter_location_by_contact = fields.Boolean(
```
Comment (max3903) [inline fieldservice_account_analytic/models/res_company.py]: @osi-scampbell I would prefix the field with `fsm_` as we could have the same with stock...

File: fieldservice_account_analytic/models/res_config_settings.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Copyright (C) 2018 - TODAY, Open Source Integrators
```
Comment (max3903) [inline fieldservice_account_analytic/models/res_config_settings.py]:
```suggestion
# Copyright (C) 2020 Open Source Integrators
```

**Review by dreispt — APPROVED** [review]
LGTM

**Review by brian10048 — APPROVED** [review]

**Review by max3903 — APPROVED** [review]

**Pull request merged**